### DISCUSSION TITLE
Customizable default controls

### DISCUSSION THREAD

**Discussion by carlosjedwab** [xyflow/xyflow#1770] [New Features]
Is there a way to customize more the controls? Like changing the zoom in and out icons, or the controls bar position inside the react flow, or even create multiple control bars with custom buttons.

**Comment by leejustin** [comment]
You should be able to customize your controls via this new(ish) feature: https://reactflow.dev/docs/api/components/controls/#extended-controls

**Reply by carlosjedwab** [reply]
Awsome! But can I change the icons os the default controls?

**Reply by moklick** [reply]
You can implement your own custom controls. I made an example for this: https://codesandbox.io/s/great-ardinghelli-k7vof?file=/src/CustomControls.js

**Reply by carlosjedwab** [reply]
Thank you!

**Reply by alexbadia1** [reply]
Hello from 2024 👋 !

The migration to v10 unifies useZoomPanHelper into the [useReactFlow hook](https://reactflow.dev/api-reference/hooks/use-react-flow). Hopefully, someone found this useful!

![image](https://github.com/xyflow/xyflow/assets/47284946/eeae8746-c06b-4499-8a74-3c445f41fa9a)

**Comment by OliverfromTriyit** [comment]
is there a way to customise just the icons without having to reimplement their default functions?
